src/modules/graph-authentication/graph-authentication.service.ts

```diff
@@ -0,0 +1,36 @@
+import * as msal from '@azure/msal-node';
+import { Inject } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import GraphAuthenticationConfig from '@ukef/config/graph-authentication.config';
+
+import { GraphAuthenticationFailedException } from './exception/graph-authentication-failed.exception';
+
+type RequiredConfigKeys = 'clientId' | 'clientSecret' | 'authority' | 'scope';
+
+export class GraphAuthenticationService {
+  private readonly cca: msal.ConfidentialClientApplication;
+  constructor(
+    @Inject(GraphAuthenticationConfig.KEY)
+    private readonly config: Pick<ConfigType<typeof GraphAuthenticationConfig>, RequiredConfigKeys>,
+  ) {
+    const clientConfig = {
+      auth: {
+        clientId: this.config.clientId,
+        authority: this.config.authority,
+        clientSecret: this.config.clientSecret,
+      },
+    };
+    this.cca = new msal.ConfidentialClientApplication(clientConfig);
+  }
+
+  getAccessToken = async (): Promise<string> => {
+    const { accessToken } = await this.cca
+      .acquireTokenByClientCredential({
+        scopes: [this.config.scope],
+      })
+      .catch(() => {
+        throw new GraphAuthenticationFailedException('Failed to get graph token');
+      });
+    return accessToken;
+  };
+}
```

Comment:
Did we consider using `@azure/identity` at all? It looks like it works really well with `@microsoft/microsoft-graph-client` and we could skip getting the access token entirely (like [so](https://learn.microsoft.com/en-gb/graph/sdks/choose-authentication-providers?tabs=Javascript#client-credentials-provider))- or do we need more control over the authentication?